File: app/src/main/java/com/OverSadBoy/samurairise/presenter/Presenter.java
package com.OverSadBoy.samurairise.presenter;

import com.OverSadBoy.samurairise.model.ModelContract;
import com.OverSadBoy.samurairise.model.database.Item;
import com.OverSadBoy.samurairise.view.ViewContract;

import java.util.ArrayList;

public class Presenter implements PresenterContract {

    private ViewContract view;
    private ModelContract model;

    public Presenter(ModelContract modelContract) {
        this.model = modelContract;
    }

    @Override
    public void attachView(ViewContract view) {
        this.view = view;
    }

    @Override
    public void detachView() {
        view = null;
    }

    @Override
    public void addAlarm(Item item) {
        model.addAlarm(item);
    }

    @Override
    public ArrayList<Item> getAlarms() {
        return model.getAlarms();
    }

    @Override
    public void onClickSwitch() {

    }

    @Override
    public void onLongClick() {

    }

    @Override
    public void addClick() {
        view.startActivityAdd();
    }

    @Override
    public void deleteClick(Item item) {
        model.deleteAlarm(item);
    }
}

File: app/src/main/java/com/OverSadBoy/samurairise/presenter/Presenter.kt
package com.OverSadBoy.samurairise.presenter

import androidx.test.runner.AndroidJUnit4
import com.OverSadBoy.samurairise.model.ModelContract
import com.OverSadBoy.samurairise.model.database.Item
import com.OverSadBoy.samurairise.view.ViewContract
import org.junit.runner.RunWith
import java.util.*

class Presenter(private val model: ModelContract?) : PresenterContract {
    private var view: ViewContract? = null
    override fun attachView(view: ViewContract?) {
        this.view = view
    }

    override fun detachView() {
        view = null
    }

    override fun addAlarm(item: Item?) {
        model.addAlarm(item)
    }

    override fun getAlarms(): ArrayList<Item?>? {
        return model.getAlarms()
    }

    override fun onClickSwitch() {}
    override fun onLongClick() {}
    override fun addClick() {
        view.startActivityAdd()
    }

    override fun deleteClick(id: Int) {
        model.deleteAlarm(id)
    }

}